src/database/models/version_item.rs

```diff
@@ -10,13 +10,62 @@ pub struct VersionBuilder {
     pub version_number: String,
     pub changelog: String,
     pub files: Vec<VersionFileBuilder>,
-    pub dependencies: Vec<(VersionId, String)>,
+    pub dependencies: Vec<DependencyBuilder>,
     pub game_versions: Vec<GameVersionId>,
     pub loaders: Vec<LoaderId>,
     pub release_channel: ChannelId,
     pub featured: bool,
 }
 
+pub struct DependencyBuilder {
+    pub project_id: Option<ProjectId>,
+    pub version_id: Option<VersionId>,
+    pub dependency_type: String,
+}
+
+impl DependencyBuilder {
+    pub async fn insert(
+        self,
+        version_id: VersionId,
+        transaction: &mut sqlx::Transaction<'_, sqlx::Postgres>,
+    ) -> Result<(), DatabaseError> {
+        let version_dependency_id = if let Some(project_id) = self.project_id {
+            sqlx::query!(
+                "
+                SELECT version.id id FROM (
+                    SELECT DISTINCT ON(v.id) v.id, v.date_published FROM versions v
+                    INNER JOIN game_versions_versions gvv ON gvv.joining_version_id = v.id AND gvv.game_version_id IN (SELECT game_version_id FROM game_versions_versions WHERE joining_version_id = $2)
+                    INNER JOIN loaders_versions lv ON lv.version_id = v.id AND lv.loader_id IN (SELECT loader_id FROM loaders_versions WHERE version_id = $2)
+                    WHERE v.mod_id = $1
+                ) AS version
+                ORDER BY version.date_published DESC
+                LIMIT 1
+                ",
+                project_id as ProjectId,
+                version_id as VersionId,
+            )
+                .fetch_optional(&mut *transaction).await?.map(|x| VersionId(x.id))
+        } else {
+            self.version_id
+        };
+
+        sqlx::query!(
+            "
+            INSERT INTO dependencies (dependent_id, dependency_type, dependency_id, mod_dependency_id)
```

Comment:
For better readability, wouldn't it better to name it `dependency_version_id` and `dependency_mod_id`?

On another note, I still think that we may need to have a semver like system for dependencies, as it will allow way more flexibility for the dependency resolver.

Reply:
Semver isn't possible, since most mods on Modrinth do not follow semver. It's not called dependency_version_id since it doesn't work that way - I can kinda explain it in discord. If dependency_id is not present, it's auto specified. It is very rare for it to be null

Reply:
That didn't answer the possible rename for better clarity